latest/docs-ref-conceptual/create-an-azure-service-principal-azure-cli.md

```diff
@@ -73,9 +73,9 @@ az ad app list --display-name MyDemoWebApp
 
 The `--display-name` option filters the returned list of apps to show those with `displayName` starting with MyDemoWebApp.
 
-### Create the service principal
+### Create the service principal with password
 
-Use [az ad sp create-for-rbac](/cli/azure/ad/sp#create-for-rbac) to create the service principal. 
+Use [az ad sp create-for-rbac](/cli/azure/ad/sp#create-for-rbac) and the `--password` parameter to create the service principal with a password. If you do not provide a value for the password parameter, the password is auto-generated. When you do not provide a role or scope, it defaults to the **Contributor** role for the current subcription.
```

Comment:
Should have been clearer; an empty value for `--password` is not accepted, but if you use the command __without__ the `--password` argument, one is autogenerated. So I would recommend changing `If you do not provide a value...` to:

> If you create a service principal without using either the `--password` or `--cert` parameter, password authentication is used and a password is generated for you.